tools/flaky-test-reporter/jsonreport/jsonreport.go

```diff
@@ -49,46 +52,44 @@ func (r *Report) writeToArtifactsDir() error {
 	return ioutil.WriteFile(outFilePath, contents, 0644)
 }
 
-func getBuildForID(buildID int) (*prow.Build, error) {
-	job := prow.NewJob(jobName, prow.PeriodicJob, "", 0)
-	if buildID == 0 {
-		latest, err := job.GetLatestBuildNumber()
-		if err != nil {
-			return nil, err
-		}
-		return job.NewBuild(latest), nil
-	} else {
-		return job.NewBuild(buildID), nil
-	}
+func getJSONForBuild(repo string, buildID int) ([]byte, error) {
+  relPath := path.Join(prow.ArtifactsDir, repo, filename)
+  job := prow.NewJob(jobName, prow.PeriodicJob, "", 0)
+  if buildID != -1 { // if buildID is specified, use it regardless of if it exists
+    return job.NewBuild(buildID).ReadFile(relPath)
+  }
+  for build := range job.GetLatestBuilds(buildCount) { // otherwise find most recent build with a report
```

Comment:
Suggest you use `GetLatestBuildNumber` instead, which is much more performant, see this example: https://github.com/knative/test-infra/blob/88a9875993cb7b24de5598265e42edf82521fe36/tools/flaky-test-reporter/result.go#L249